iroha/src/merkle.rs

```diff
@@ -15,14 +15,12 @@ impl MerkleTree {
         }
     }
 
-    /// Builds a Merkle Tree from sorted array of `ValidBlocks`.
-    //TODO: should we check or sort blocks here?
-    pub fn build(&mut self, blocks: &[&ValidBlock]) {
-        //hm, can we write map(ValidBlock::hash) in Rust?
-        let mut nodes: std::collections::VecDeque<Node> = blocks
-            .iter()
-            .map(|block| Node::Leaf { hash: block.hash() })
-            .collect();
+    /// Builds a Merkle Tree from an array of `Hash` values of `Block` or `Transaction`.
+    pub fn build(&mut self, hashes: &[Hash]) {
+        let mut hashes: Vec<Hash> = hashes.to_vec();
+        hashes.sort();
```

Comment:
May we inline this operations below?

Reply:
We need `itertools::sorted` for this should I add this crate to our dependencies? https://docs.rs/itertools/0.9.0/itertools/trait.Itertools.html#method.sorted

Reply:
no, let's skip it then